### PR TITLE
Newlookwithoutssl

### PR DESCRIPTION
**Pull request by krishnatejaperannagari**
using submodule without the need of adding apache ssi to the cgi scripts

### CODE REVIEW

File: buildfarm/web/__init__.py [hunk 1]
```diff
@@ -330,6 +330,38 @@ def print_log_cc_checker(input):
     return output
 
 
+def log_collapsible_html(type, title, output, id, status=""):
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: Why is this introducing a new function, rather than modifying the exsting one - make_collapsible_html?

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "web/samba-web"]
+	path = web/samba-web
+	url = https://github.com/krishnatejaperannagari/samba-web
```
Comment (jelmer) [inline .gitmodules]: Please either use the standard samba-we repository or propose changes to it that are suitable.

Comment (jelmer) [inline .gitmodules]: Perhaps you can introduce a modified version of FileLoad for loading samba-web include files that transforms any relative links to be based on https://www.samba.org/?

File: web/about.html [hunk 3]
```diff
@@ -1,8 +1,60 @@
...
+<div class="noPrint" id="nav">
+	<ul>
+	        <li><a href="http://build.samba.org/">Build Samba</a></li>
+                <ul>
+                    <li><a href="/about.html">About the Build Farm</a></li>
+                    <li><a href="/instructions.html">Adding a new machine</a></li>
+                    <li><a href="http://www.samba.org/">Samba Web pages</a></li>
+                </ul>
+         <br>       
+		<!--#include virtual="/samba-web/menu_think_samba_closed.html" -->
```
Comment (jelmer) [inline web/about.html]: This still seems to use ssi?

Comment (krishnatejaperannagari) [inline web/about.html]: reviews made and pushed to the branch newlookwithoutssl

On Tue, Jul 15, 2014 at 10:20 PM, Jelmer Vernooij notifications@github.com
wrote:

> In web/about.html:
> 
> > -           <div id="bd">
> > -               <div id="bdCols">
> >   +
> >   +
> >   +
> >   +
> >   +<div class="noPrint" id="nav">
> > -   <ul>
> > -           <li><a href="http://build.samba.org/">Build Samba</a></li>
> > -                <ul>
> > -                    <li><a href="/about.html">About the Build Farm</a></li>
> > -                    <li><a href="/instructions.html">Adding a new machine</a></li>
> > -                    <li><a href="http://www.samba.org/">Samba Web pages</a></li>
> > -                </ul>
> > -         <br>
> > -       
> 
> This still seems to use ssi?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jelmer/build-farm/pull/10/files#r14947599.

## 

krishna teja perannagari

File: buildfarm/util.py [hunk 4]
```diff
@@ -40,6 +42,19 @@ def FileLoad(filename):
     finally:
         f.close()
 
+def LoadAndParse(filename):
```
Comment (jelmer) [inline buildfarm/util.py]: Please give this a clearer name. Perhaps SambaWebFileLoad ?

File: buildfarm/util.py [hunk 5]
```diff
@@ -40,6 +42,19 @@ def FileLoad(filename):
...
+    """loads file and changes the links to suit buildfarm""" 
+    f = open(filename, 'r')
+    try:
+        links = f.read()
+    finally:
+        f.close()
+    modifiedlinks = ""
+    for line in links.splitlines():
+        line = re.sub('href="/samba', 'href="http://www.samba.org/samba' ,line)
+        modifiedlinks += line + "\n"
```
Comment (jelmer) [inline buildfarm/util.py]: += to create strings is really slow, since each call reallocates the string. Is there any reason you're doing this per line, rather than just calling re.sub() on links?

File: buildfarm/util.py [hunk 6]
```diff
@@ -40,6 +42,19 @@ def FileLoad(filename):
     finally:
         f.close()
 
+def LoadAndParse(filename):
+    """loads file and changes the links to suit buildfarm""" 
+    f = open(filename, 'r')
+    try:
+        links = f.read()
```
Comment (jelmer) [inline buildfarm/util.py]: links is a confusing name, since this is the full text of the file. What about "text" ?

File: buildfarm/util.py [hunk 7]
```diff
@@ -40,6 +42,19 @@ def FileLoad(filename):
...
+    finally:
+        f.close()
+    modifiedlinks = ""
+    for line in links.splitlines():
+        line = re.sub('href="/samba', 'href="http://www.samba.org/samba' ,line)
```
Comment (jelmer) [inline buildfarm/util.py]: space should be after the comma, rather than before

File: buildfarm/web/__init__.py [hunk 8]
```diff
@@ -266,7 +266,7 @@ def pretty_print(self, log):
         if not buf == "":
             divhtml = "".join(make_collapsible_html('testlinks', 'Shortcut to failed tests', "<a name='shortcut2errors'></a>%s" % buf, self.indice, ""))+"\n"
             log = re.sub("Running action\s+test", divhtml, log)
-        return "<pre>%s</pre>" % log
+        return "<pre>%s </pre>" % log
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: unnecessary change

File: buildfarm/web/__init__.py [hunk 9]
```diff
@@ -331,7 +331,7 @@ def print_log_cc_checker(input):
 
 
 def make_collapsible_html(type, title, output, id, status=""):
-    """generate html for a collapsible section
+    """generate html for a collapsible section of log as it is already in pre
```
Comment (jelmer) [inline buildfarm/web/__init__.py:346]: why is it already in pre? The extra uses of pre seem unnecessary to me, they're all empty.

File: buildfarm/web/__init__.py [hunk 10]
```diff
@@ -971,22 +1014,46 @@ def html_page(self, form, lines):
...
 
-        yield util.FileLoad(os.path.join(webdir, "header2.html"))
-
         tree = get_param(form, "tree")
         host = get_param(form, "host")
         compiler = get_param(form, "compiler")
         yield "".join(self.main_menu(tree, host, compiler))
-        yield util.FileLoad(os.path.join(webdir, "header3.html"))
+        yield util.FileLoad(os.path.join(webdir, "bannernav1.html"))
+        yield util.LoadAndParse(os.path.join(webdir, "samba-web", "menu_think_samba_closed.html"))
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: \o/ This is much cleaner, thanks!

File: buildfarm/web/__init__.py (hunk 9 above)
Comment (krishnatejaperannagari) [inline buildfarm/web/__init__.py:346]: see the function     def pretty_print(self, log):

line 269: return "<pre>%s</pre>" % log
actually log already contains pre header due to the make_collapsible html  substiting the part of log with the colapsible html thats contais <pre> so i am closing the already exising one in log_collapsible_html
this is necessary to place parts of log uneffected by collapsible htmls to be in <pre>
this (displaying even the log parts that dont match the regularexpression search) is rectified in the current branch improvedereporting.

On Fri, Jul 18, 2014 at 4:12 AM, Jelmer Vernooij notifications@github.com
wrote:

> In buildfarm/web/**init**.py:
> 
> > @@ -331,7 +331,7 @@ def print_log_cc_checker(input):
> > 
> >  def make_collapsible_html(type, title, output, id, status=""):
> > -    """generate html for a collapsible section
> > -    """generate html for a collapsible section of log as it is already in pre
> 
> why is it already in
> 
> ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jelmer/build-farm/pull/10/files#r15088856.

## 

krishna teja perannagari

File: buildfarm/web/__init__.py [hunk 11]
```diff
@@ -362,7 +371,7 @@ def web_paths(t, paths):
     if t.scm == "git":
         ret = ""
         for path in paths:
-            ret += " <a href=\"%s/?p=%s;a=history;f=%s%s;h=%s;hb=%s\">%s</a>" % (GITWEB_BASE, t.repo, t.subdir, path, t.branch, t.branch, path)
+            ret += " <a href=\"%s/?p=%s;a=history;f=%s%s;h=%s;hb=%s\">%s</a>&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp&nbsp" % (GITWEB_BASE, t.repo, t.subdir, path, t.branch, t.branch, path)
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: &nbsp here is missing a closing ";"

What is the purpose of the non-breaking spaces here? &nbsp; shouldn't be used for layout - please use CSS for that.

File: buildfarm/web/__init__.py [hunk 12]
```diff
@@ -428,7 +437,7 @@ def show_oldrevs(self, myself, build, host, compiler, limit=None):
                 build_link(myself, old_build),
                 util.dhm_time(old_build.age))
 
-        yield "</tbody></table>\n"
+        yield "</tbody></table><br>\n"
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: Why is <br> necessary here? the next line already starts a new paragraph.

In general, please stick to modern HTML and e.g. use <br/> rather than <br>

File: buildfarm/web/__init__.py [hunk 13]
```diff
@@ -522,15 +531,19 @@ def render(self, myself, build, plain_logs=False, limit=10):
             # allow them to wrap in some way?
             if err == "":
                 yield "<h2>No error log available</h2>\n"
+                yield "<br>"
             else:
                 yield "<h2>Error log:</h2>"
-                yield "".join(make_collapsible_html('action', "Error Output", "\n%s" % err, "stderr-0", "errorlog"))
+                yield "".join(make_collapsible_html('action', "Error Output", "\n%s\n" % err, "stderr-0", "errorlog"))
```
Comment (jelmer) [inline buildfarm/web/__init__.py:542]: Why is the extra newline necessary ?

File: buildfarm/web/__init__.py [hunk 14]
```diff
@@ -881,7 +918,7 @@ def render(self, myself, tree, revision):
         changes = branch.changes_summary(revision)
         yield "".join(self.history_row_html(myself, entry, t, changes))
         diff = highlight(diff, DiffLexer(), HtmlFormatter())
-        yield "<pre>%s</pre>\n" % diff.encode("utf-8")
+        yield "<pre>%s</pre>" % diff.encode("utf-8")
```
Comment (jelmer) [inline buildfarm/web/__init__.py:925]: Why remove the extra newline?

File: buildfarm/web/__init__.py [hunk 15]
```diff
@@ -943,22 +984,26 @@ def main_menu(self, tree, host, compiler):
...
         for h in self.buildfarm.hostdb.hosts():
-            host_dict[h.name] = "%s -- %s" % (h.platform.encode("utf-8"), h.name)
+                 host_dict[h.name] = "%s-%s" % (h.platform.encode("utf-8"), h.name)
         yield "".join(select("host", host_dict, default=host))
+        yield "<br/>\n<br/>\n"
+
         tree_dict = {}
         for t in self.buildfarm.trees.values():
-            tree_dict[t.name] = "%s:%s" % (t.name, t.branch)
+                tree_dict[t.name] = "%s:%s" % (t.name, t.branch)
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: Please use 4-space indents, per PEP8 and consistent with the rest of the file.

File: buildfarm/web/__init__.py [hunk 16]
```diff
@@ -943,22 +984,26 @@ def main_menu(self, tree, host, compiler):
         """main page"""
 
         yield "<form method='GET'>\n"
-        yield "<div id='build-menu'>\n"
+        yield "<div id='newbuildmenu'>\n"
         host_dict = {}
         for h in self.buildfarm.hostdb.hosts():
-            host_dict[h.name] = "%s -- %s" % (h.platform.encode("utf-8"), h.name)
+                 host_dict[h.name] = "%s-%s" % (h.platform.encode("utf-8"), h.name)
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: Please use 4-space indents, per PEP8 and consistent with the rest of the file.

File: buildfarm/web/__init__.py [hunk 17]
```diff
@@ -971,22 +1016,46 @@ def html_page(self, form, lines):
...
         yield "".join(self.main_menu(tree, host, compiler))
-        yield util.FileLoad(os.path.join(webdir, "header3.html"))
+        yield util.FileLoad(os.path.join(webdir, "bannernav1.html"))
+        yield util.SambaWebFileLoad(os.path.join(webdir, "samba-web", "menu_think_samba_closed.html"))
+        yield util.SambaWebFileLoad(os.path.join(webdir, "samba-web", "menu_get_samba_closed.html"))
+        yield util.SambaWebFileLoad(os.path.join(webdir, "samba-web", "menu_learn_samba_closed.html"))
+        yield util.SambaWebFileLoad(os.path.join(webdir, "samba-web", "menu_talk_samba_closed.html"))
+        yield util.SambaWebFileLoad(os.path.join(webdir, "samba-web", "menu_hack_samba_closed.html"))
+        yield util.SambaWebFileLoad(os.path.join(webdir, "samba-web", "menu_contact_samba_closed.html"))
+        yield util.FileLoad(os.path.join(webdir, "bannernav2.html"))
```
Comment (jelmer) [inline buildfarm/web/__init__.py:1046]: Can bannernav2.html not be loaded from samba-web?

File: buildfarm/web/__init__.py [hunk 18]
```diff
@@ -971,22 +1016,46 @@ def html_page(self, form, lines):
...
         yield "".join(lines)
-        yield util.FileLoad(os.path.join(webdir, "footer.html"))
-        yield "</body>"
-        yield "</html>"
+        yield util.FileLoad(os.path.join(webdir, "sitemap1.html"))
```
Comment (jelmer) [inline buildfarm/web/__init__.py]: Can these various sitemap files not be loaded from samba-web?

**Comment by jelmer** [comment]
Can you rebase this branch and the other?

File: buildfarm/web/__init__.py (hunk 13 above)
Comment (krishnatejaperannagari) [inline buildfarm/web/__init__.py:542]: so that the scrollbar doesnt touch the text. an empty line will provide some gap between scrollbar and log

File: buildfarm/web/__init__.py (hunk 14 above)
Comment (krishnatejaperannagari) [inline buildfarm/web/__init__.py:925]: it is not necessary in new layout

File: buildfarm/web/__init__.py (hunk 17 above)
Comment (krishnatejaperannagari) [inline buildfarm/web/__init__.py:1046]: the related sites in bannernav2 and the sitemap can be loadede from samba web only if u can change the sambaweb pages to seperate the content related to related sites and sitemap from index.html and other pages and use virtual headers to load them:

the following files to be added some problem since github has parsed my text as markdown will send as attachments along with the places in original buildfarm where u need to change them.

menu_relatedsites_samba_closed.html:
                                <h4>Related Sites</h4>
                                                                <br><br>

```
                            <ul>
                                <li><a href="http://cwrap.org/">cwrap.org</a></li>
                                <li><a href="http://linux-cifs.samba.org/">linux-cifs.samba.org</a></li>
                                <li><a href="http://talloc.samba.org/">talloc.samba.org</a></li>
                                <li><a href="http://tevent.samba.org/">tevent.samba.org</a></li>
                                <li><a href="http://tdb.samba.org/">tdb.samba.org</a></li>
                                <li><a href="http://ldb.samba.org/">ldb.samba.org</a></li>
                                <li><a href="http://jcifs.samba.org/">jcifs.samba.org</a></li>
                                <li><a href="http://rsync.samba.org/">rsync.samba.org</a></li>
                                <li><a href="http://ccache.samba.org/">ccache.samba.org</a></li>
                            <li><a href="http://ctdb.samba.org/">ctdb.samba.org</a></li>
                                <li><a href="http://ppp.samba.org/">ppp.samba.org</a></li>
                            </ul>
```

menu_sitemap_samba_closed.html:
                    <div class="noPrint" id="sitemap">
                        <div id="sitemapContent">
                            <div class="sitemapList">
                                <h5>Think Samba</h5>
                                
                            </div>
                            <div class="sitemapList">
                                <h5>Get Samba</h5>
                                
                            </div>
                            <div class="sitemapList">
                                <h5>Learn Samba</h5>
                                
                            </div>
                            <div class="clearing"></div>

```
                        <div class="sitemapList">
                            <h5>Talk Samba</h5>
                            <!--#include virtual="/samba/menu_talk_samba_raw.html" -->
                        </div>
                        <div class="sitemapList">
                            <h5>Hack Samba</h5>
                            <!--#include virtual="/samba/menu_hack_samba_raw.html" -->
                        </div>
                        <div class="sitemapList">
                            <h5>Contact Samba</h5>
                            <!--#include virtual="/samba/menu_contact_samba_raw.html" -->
                        </div>
                        <div class="clearing"></div>
                    </div>
                    <div id="sitemapBottom"></div>
                </div>
```

see my recent commit changes to sambaweb and content it uses these files from samba web

File: buildfarm/util.py [hunk 19]
```diff
@@ -40,6 +43,40 @@ def FileLoad(filename):
...
+    text = re.sub('href="/samba', 'href="http://www.samba.org/samba', text)
+    return text
+
+class Sitemap(object):
+
+    def __init__(self, webdir):
+        self.webdir = webdir
+        
+    def file_load(self, filename):
+        self.filename = os.path.join(self.webdir, filename)
```
Comment (jelmer) [inline buildfarm/util.py]: Please don't assign member instance variables if they're not going to be used outside of a function. Rather, use a local variable (perhaps "path"?)

File: buildfarm/util.py [hunk 20]
```diff
@@ -40,6 +43,40 @@ def FileLoad(filename):
...
+        f = open(self.filename, 'r')
+        try:
+            text = f.read()
+        finally:
+            f.close()
+        text = re.sub('<!--#include virtual="/samba/(.*)" -->', self.add_virtual_headers, text)
```
Comment (jelmer) [inline buildfarm/util.py]: Thanks so much, I think this is a step in the right direction.

It seems like this should just be a function with a nested function "add_virtual_headers" rather than a class, since there's nothing particularly in the class.

In fact, you could also make SambaWebFileLoad take two arguments: a webdir and a filename.

In that case, you could replace self.add_virtual_headers here with "lambda filename: SambaWebFileLoad(webdir, m.group(1))".

File: web/right2.html [hunk 21]
```diff
@@ -0,0 +1,20 @@
+                <br>
+		<p class="headline">Release History</p>
```
Comment (jelmer) [inline web/right2.html]: Can we perhaps just get rid of these boxes on the right hand side? They don't really add anything for the buildfarm.

Comment (jelmer) [inline web/right2.html]: Same for the boxes in right1.html.

**Comment by jelmer** [comment]
You're adding various icons. Where are they used?

**Comment by krishnatejaperannagari** [comment]
they are used in the place of the icons replaced i.e in the collapsiblehtml
to suit the new look

On Tue, Aug 5, 2014 at 6:42 AM, Jelmer Vernooij notifications@github.com
wrote:

> You're adding various icons. Where are they used?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jelmer/build-farm/pull/10#issuecomment-51138695.

## 

krishna teja perannagari

File: buildfarm/util.py (hunk 20 above)
Comment (krishnatejaperannagari) [inline buildfarm/util.py]: done added add_virtual headers within the sambawebfileload

On Tue, Aug 5, 2014 at 6:39 AM, Jelmer Vernooij notifications@github.com
wrote:

> In buildfarm/util.py:
> 
> > -    text = re.sub('href="/samba', 'href="http://www.samba.org/samba
> >   ', text)
> > -    return text
> >   +
> >   +class Sitemap(object):
> >   +
> > -    def **init**(self, webdir):
> > -        self.webdir = webdir
> >   +
> > -    def file_load(self, filename):
> > -        self.filename = os.path.join(self.webdir, filename)
> > -        f = open(self.filename, 'r')
> > -        try:
> > -            text = f.read()
> > -        finally:
> > -            f.close()
> > -        text = re.sub('', self.add_virtual_headers, text)
> 
> It seems like this should just be a function with a nested function
> "add_virtual_headers" rather than a class, since there's nothing
> particularly in the class.
> 
> In fact, you could also make SambaWebFileLoad take two arguments: a webdir
> and a filename.
> 
> In that case, you could replace self.add_virtual_headers here with "lambda
> filename: SambaWebFileLoad(webdir, m.group(1))".
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jelmer/build-farm/pull/10/files#r15790920.

## 

krishna teja perannagari

File: web/right2.html (hunk 21 above)
Comment (krishnatejaperannagari) [inline web/right2.html]: they are added to suit the new look. as an alternative i have replaced the
right part with footer.html in samba_web see the commit modifications to
use footer and search field

On Tue, Aug 5, 2014 at 6:41 AM, Jelmer Vernooij notifications@github.com
wrote:

> In web/right2.html:
> 
> > @@ -0,0 +1,20 @@
> > -                <br>
> > -       <p class="headline">Release History</p>
> 
> Same for the boxes in right1.html.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jelmer/build-farm/pull/10/files#r15790957.

## 

krishna teja perannagari

File: buildfarm/util.py (hunk 20 above)
Comment (jelmer) [inline buildfarm/util.py]: On Tue, Aug 05, 2014 at 06:39:17AM -0700, krishnatejaperannagari wrote:

> > -    text = re.sub('href="/samba', 'href="http://www.samba.org/samba', text)
> > -    return text
> >   +
> >   +class Sitemap(object):
> >   +
> > -    def **init**(self, webdir):
> > -        self.webdir = webdir
> > -   
> > -    def file_load(self, filename):
> > -        self.filename = os.path.join(self.webdir, filename)
> > -        f = open(self.filename, 'r')
> > -        try:
> > -            text = f.read()
> > -        finally:
> > -            f.close()
> > -        text = re.sub('', self.add_virtual_headers, text)
> 
> done added add_virtual headers within the sambawebfileload

Awesome, thanks! :)
